### PR TITLE
Adding version to GetSite. Fixes #1001

### LINKED ISSUE #1001: Add a site version API
### Is your proposal related to a problem?

In a perfect world, API changes would never require special handling, but in practice there will likely be many different versions of Lemmy installed and operating on the Fediverse.  Additionally, there will likely be a variety of third party clients used to connect to these instances.  In order for clients to support as many instances as possible, it would be helpful if Lemmy provided a standardized API for clients and peer instances to request the site version, client-server API version, server-server API version, and perhaps other information about installed extensions.

### Describe the solution you'd like

The Matrix protocol includes [a fairly extensible version API](https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-versions) which remains pretty simple to implement - especially if you are only trying to support one version in your implementation (as the Lemmy server likely will for some time). None the less, it allows for clients and servers which support multiple API implementations to choose an appropriate version, or enable/disable features as appropriate.

Additionally, it might be nice to have an optional `extensions` key included in this response. An array of strings to indicate extra features supported by the instance. These may include captcha protected logins, custom emoji sets, additional contact/info fields for the user page, or anything the Lemmy community can think up, really.

### Describe alternatives you've considered

* Never change the API /s

* Include version info in the SiteView structure which is already transmitted (though this requires SiteView to maintain backwards compatibility)

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

File: docs/src/contributing_websocket_http_api.md [hunk 1]
```diff
@@ -754,6 +754,8 @@ Search types are `All, Comments, Posts, Communities, Users, Url`
     site: Option<SiteView>,
     admins: Vec<UserView>,
     banned: Vec<UserView>,
+    online: usize,
```
Comment (Nutomic) [inline docs/src/contributing_websocket_http_api.md:757]: This is still broken right? Should mention that here.

**Comment by Nutomic** [comment]
We should also remove `ui/src/version.ts` at the same time, to simplify the release script and avoid cases where the wrong version number is shown due to caching.

File: docs/src/contributing_websocket_http_api.md (hunk 1 above)
Comment (dessalines) [reply docs/src/contributing_websocket_http_api.md:757]: Done.

**Comment by dessalines** [comment]
> We should also remove ui/src/version.ts at the same time, to simplify the release script and avoid cases where the wrong version number is shown due to caching.

Good call, I'll do this soon.

**Comment by dessalines** [comment]
Okay added this. One kinda scary thing tho, is that the nginx sometimes incorrectly caches our front end I've noticed, so you the version might say `0.7.1`, when really the front end you're looking at is a cached older version.

I'm not exactly sure why our nginx setup does this tho.

**Comment by Nutomic** [comment]
Are you sure its nginx and not your browser? `proxy_cache` is commented out in our prod nginx config, so it shouldnt cache anything.

Though I see that we have it enabled in this repo, should really comment it out or remove it.

https://github.com/LemmyNet/lemmy/blob/main/ansible/templates/nginx.conf#L70

**Comment by dessalines** [comment]
Ya I don't think its the proxy_cache, it might be this:

`static ref CACHE_CONTROL_VALUE: String = format!("public, max-age={}", 60 * 60);` in main.rs

**Comment by Nutomic** [comment]
Thats for the cache-control header (used by the browser). We can remove it, but I assumed that it still checks the etag or last-modified to avoid that problem. Maybe we could tell yarn to name compiled files based on their hash, so if the content changes, it is also under a different url.

**Comment by dessalines** [comment]
Here's the headers at least:

image

On page refresh it seems to work, but sometimes on mobile I'll go back to the browser and it'll be like 10 versions behind. I'm not sure what's going on.

**Comment by Nutomic** [comment]
Okay we can disable the caching then, anyway bandwidth is not a problem on the server side.

**Pull request merged**